src/struct.cpp

```diff
@@ -173,6 +173,11 @@ void StructManager::Add(const std::string &name,
   struct_map_[name] = std::make_unique<Struct>(size, allow_override);
 }
 
+void StructManager::Add(const std::string &name, Struct &&record)
+{
+  struct_map_[name] = std::make_unique<Struct>(record);
```

Comment:
Also, it looks like this could be changed to `make_shared`, to be clearer about what it's doing and it'd be a bit more efficient with its allocations.